### ISSUE TITLE
FR: Manga: RTL in two column, landscape view

### ISSUE THREAD

**Issue by AstralH3X** [koreader/koreader#11085]
Currently there's no way (that I can find) to have two column, right to left reading. I've tried epubs, cbz, etc. I have a Kobo Elipsa which would be *fantastic* for manga with both pages, except that I as of yet can't find anyway to do this without pages being flipped, with page 1 on the left, and page 2 on the right.

**Describe the solution you'd like**
An option which literally just flips which page is on the left, and which is on the right, and *optionally* an option to offset odds/evens so that spreads line up. (Ie, if a book has an extra page that throws off odds/evens, to offset it by one so that spreads are matched up)

**Describe alternatives you've considered**
If there's literally any sane way to do this myself I guess I'd be happy but "Open up every page of every manga ever in Photoshop and stitch them together" is the only option I've seen so far.

**Comment by mergen3107** [comment]
Cross-ref #10204 for landscape. 

For RTL, there are Top menu settings for changing directions. Read the manual

**Comment by AstralH3X** [comment]
That just flips whether tapping on the left or the right triggers next page vs previous page. It does not change which page is displayed on the left vs right.

**Comment by mergen3107** [comment]
For this, use KCC to properly arrange manga. 

KOReader is for reading books, not making them.

**Comment by AstralH3X** [comment]
I did use KCC, with Manga Mode enabled, and when inspecting the epub I can verify the line, 
  <spine page-progression-direction="rtl" toc="ncx">
Are there settings within KCC that I need to verify that would fix this behavior?

**Comment by mergen3107** [comment]
Oh, don't use epubs for manga. CBZ is the only way to enjoy it :D

**Comment by AstralH3X** [comment]
CBZ does not have the two page layout, so far as I can tell, and that's specifically what I am trying to accomplish. Two page, RTL, landscape view.

**Comment by hugleo** [comment]
If the pages are 1.jpg 2.jpg 3.jpg 4.jpg... Just create a script to invert the names. 1.jpg rename to 2.jpg. 2.jpg rename to 1.jpg, after that use your already know stuff to flip.

**Comment by mergen3107** [comment]
You will have much better chances just stitching them yourself with ImageMagick, than waiting for two-page landscape here :D

**Comment by AstralH3X** [comment]
Is it really that difficult to just flip whether a page is rendered on the left or right?

**Comment by hius07** [comment]
cbz are rendered by muPDF, double-page layout request is https://github.com/koreader/koreader/issues/10204.
epub are rendered by crengine. Switching and shifting pages are doable I think.

**Comment by poire-z** [comment]
> epub are rendered by crengine. Switching and shifting pages are doable I think.

Doable... maybe, but painful. #9323 https://github.com/koreader/koreader/issues/7082#issuecomment-752731107

**Comment by hius07** [comment]
I meant manga (pics) only. Better than nothing.

**Comment by NiLuJe** [comment]
> You will have much better chances just stitching them yourself with ImageMagick, than waiting for two-page landscape here :D

Yup, that's [exactly](http://svn.ak-team.com/svn/Bash/trunk/CBZ_explode.sh) what [I do](http://svn.ak-team.com/svn/Bash/trunk/CBZ_gen.sh).

**Comment by whcwhc123** [comment]
yes，i also want that........
![IMG_20240404_201852](https://github.com/koreader/koreader/assets/76554352/ae3baa39-42f7-4f88-83ac-34d0b1b47200)

**Comment by whcwhc123** [comment]
> Maybe a gesture like double tap that can advance 3 pages. For example, if you're on page 1, a double tap will advance you to page 4. Use a single left tap to go to the previous page, or invert the single tap behavior.

no， the pages they are all correct，"1and 2" then "3 and 4"，they are all right，but for japanse manga, it is deiffent，it should "2 and 1" then "4 and 3".....

**Comment by whcwhc123** [comment]
![IMG_20240501_122256](https://github.com/koreader/koreader/assets/76554352/9184e44e-8e47-49b9-a67f-a95ecd6c7e9b)
![IMG_20240501_122250](https://github.com/koreader/koreader/assets/76554352/722b43fb-36dc-47c6-9f75-8bb41b60a865)
I find in the “zhangyue”app，it can change left and right。I thought maybe it is a simple thing  ，koreader please update this function:-)

**Comment by AjeetAthwal** [comment]
Just wanted to add that I'd love this too. I don't see a 2 page option for CBZ files at all.

**Comment by AjeetAthwal** [comment]
Just wanted to add that I'd love this too. I don't see a 2 page option for CBZ files at all.

**Comment by ssbroad** [comment]
If you want to turn on the page turn animation to read comics, then epub is the only choice instead of cbz.
Unfortunately, epub reading comics requires the function mentioned in the title, and koreader currently has no plans to improve it.

**Comment by ssbroad** [comment]
> Oh no, please don't use epubs for manga. CBZ is the only way to enjoy it :D

If you want to turn on the page turn animation to read comics, then epub is the only choice instead of cbz.
Unfortunately, epub reading comics requires the function mentioned in the title, and koreader currently has no plans to improve it.

**Comment by ssbroad** [comment]
> Oh no, please don't use epubs for manga. CBZ is the only way to enjoy it :D

If you want to turn on the page turn animation to read comics, then epub is the only choice instead of cbz.
Unfortunately, epub reading comics requires the function mentioned in the title, and koreader currently has no plans to improve it.

**Comment by OGKevin** [comment]
👋🏾 Since the MR didn’t make it upstream, I’ve extracted it into a separate plugin.

It’s still early days, but it seems to work on my machine.
Feel free to give it a test, and please make sure to make backups before you do.

If you encounter any bugs or have feedback, please open an issue. 🙂

https://github.com/OGKevin/comicreader.koplugin
